File: CS/WindowsFormsApplication25/CustomBreadCrumbEditViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsFormsApplication25 {
    public class CustomBreadCrumbEditViewInfo : BreadCrumbEditViewInfo {
        public CustomBreadCrumbEditViewInfo(RepositoryItem item)
            : base(item) {
        }
        protected override void UpdateNodeState(BreadCrumbNodeInfo nodeInfo) {
            if(!OwnerControl.Enabled) {
                nodeInfo.State = ObjectState.Disabled;
                return;
            }
            if(OwnerControl.IsDesignMode)
                return;
            if(ExpandedNode.NodeInfo != null && ExpandedNode.NodeInfo.IsMatch(nodeInfo)) {
                nodeInfo.State = ObjectState.Hot | ObjectState.Pressed;
                return;
            }
            if(HotNode.NodeInfo != null && HotNode.NodeInfo.IsMatch(nodeInfo)) {
                nodeInfo.State = ObjectState.Hot;
                //--- Comment this code to disable pressed state
                if((OwnerEdit.Properties as RepositoryItemCustomBreadCrumbEdit).AllowDropDownButtonSelectedState) {
                    if((Control.MouseButtons & MouseButtons.Left) != 0) {
                        nodeInfo.State |= ObjectState.Pressed;
                    }
                }
                //---
            } else {
                nodeInfo.State &= ~ObjectState.Hot;
            }
        }
    }
}

File: VB/WindowsFormsApplication25/CustomBreadCrumbEditViewInfo.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsFormsApplication25
    Public Class CustomBreadCrumbEditViewInfo
        Inherits BreadCrumbEditViewInfo

        Public Sub New(ByVal item As RepositoryItem)
            MyBase.New(item)
        End Sub
        Protected Overrides Sub UpdateNodeState(ByVal nodeInfo As BreadCrumbNodeInfo)
            If Not OwnerControl.Enabled Then
                nodeInfo.State = ObjectState.Disabled
                Return
            End If
            If OwnerControl.IsDesignMode Then
                Return
            End If
            If ExpandedNode.NodeInfo IsNot Nothing AndAlso ExpandedNode.NodeInfo.IsMatch(nodeInfo) Then
                nodeInfo.State = ObjectState.Hot Or ObjectState.Pressed
                Return
            End If
            If HotNode.NodeInfo IsNot Nothing AndAlso HotNode.NodeInfo.IsMatch(nodeInfo) Then
                nodeInfo.State = ObjectState.Hot
                '--- Comment this code to disable pressed state
                If (TryCast(OwnerEdit.Properties, RepositoryItemCustomBreadCrumbEdit)).AllowDropDownButtonSelectedState Then
                    If (Control.MouseButtons And MouseButtons.Left) <> 0 Then
                        nodeInfo.State = nodeInfo.State Or ObjectState.Pressed
                    End If
                End If
                '---
            Else
                nodeInfo.State = nodeInfo.State And Not ObjectState.Hot
            End If
        End Sub
    End Class
End Namespace
